test/powershell/Modules/Microsoft.PowerShell.Utility/Export-Csv.Tests.ps1

```diff
@@ -43,81 +45,154 @@ Describe "Export-Csv" -Tags "CI" {
     }
-}
 
-Describe "Export-Csv DRT Unit Tests" -Tags "CI" {
-    BeforeAll {
-        $filePath = Join-Path $TestDrive -ChildPath "test.csv"
-        $newLine = [environment]::NewLine
+    It "Should support -LiteralPath parameter" {
+        $testObject | Export-Csv -LiteralPath $testCsv
+        $results = Import-Csv -Path  $testCsv
+
+        $results | Should -HaveCount 3
+    }
+
+    It "Should overwrite file without -NoClobber parameter" {
+        $P1 | Export-Csv -Path $testCsv
+        $P2 | Export-Csv -Path $testCsv
+        $results = Import-Csv -Path $testCsv
+
+        $results.P2 | Should -BeExactly "second"
+        $property = $results.PSObject.Properties.Name
+        $property | Should -BeExactly "P2"
+    }
+
+    It "Should not overwrite file with -NoClobber parameter" {
+        $P1 | Export-Csv -Path $testCsv
+        { $P2 | Export-Csv -Path $testCsv -NoClobber} | Should -Throw -ErrorId "NoClobber,Microsoft.PowerShell.Commands.ExportCsvCommand"
+        $results = Import-Csv -Path $testCsv
+
+        $results.P1 | Should -BeExactly "first"
+        $property = $results.PSObject.Properties.Name
+        $property | Should -BeExactly "P1"
+    }
+
+    It "Should not overwrite read-only file without -Force parameter" {
+        $P1 | Export-Csv -Path $testCsv
+        Set-ItemProperty -Path $testCsv -Name IsReadOnly -Value $true
+
+        { $P2 | Export-Csv -Path $testCsv } | Should -Throw -ErrorId "FileOpenFailure,Microsoft.PowerShell.Commands.ExportCsvCommand"
+        $results = Import-Csv -Path $testCsv
+
+        $results.P1 | Should -BeExactly "first"
+        $property = $results.PSObject.Properties.Name
+        $property | Should -BeExactly "P1"
+    }
+
+    It "Should overwrite read-only file with -Force parameter" {
+        $P1 | Export-Csv -Path $testCsv
+        Set-ItemProperty -Path $testCsv -Name IsReadOnly -Value $true
+
+        $P2 | Export-Csv -Path $testCsv -Force
+        $results = Import-Csv -Path $testCsv
+
+        $results.P2 | Should -BeExactly "second"
+        $property = $results.PSObject.Properties.Name
+        $property | Should -BeExactly "P2"
+    }
+
+    It "Should not export to file if -WhatIf parameter specified" {
+        $P1 | Export-Csv -Path $testCsv -WhatIf
+        $testCsv | Should -Not -Exist
+    }
+
+    It "Should append to file if -Append parameter specified" {
+        $P1 | Export-Csv -Path $testCsv
+        $P11 | Export-Csv -Path $testCsv -Append
+        $results = Import-Csv -Path $testCsv
+
+        $results[0].P1 | Should -BeExactly "first"
+        $results[1].P1 | Should -BeExactly "eleventh"
+        $property = $results[0].PSObject.Properties.Name
+        $property | Should -BeExactly "P1"
+    }
+
+    It "Should append to empty file if -Append parameter specified" {
```

Comment:
I believe we should remove the tests - it repeats the previous test.

Reply:
There is a branch in code that this test covers.

Reply:
Please add the comment that it does not duplicate the previous test and checks another branch of the code.

Reply:
Added.

Reply:
Closed.